### PR TITLE
KP-624 - [EC2] Refactor to use Kaholo AWS Plugin

### PR DESCRIPTION
**Pull request by kdebowski-kaholo**

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "execProgram": "node",
   "exposeRoute": false,
   "imgUrl": "logo.png",
-  "version": "2.8.0",
+  "version": "3.0.0",
```
Comment (graddaniel) [inline config.json:8]: Refactors by nature don't introduce new features, so this is a new patch at most - `2.8.1`

**Review by graddaniel — CHANGES_REQUESTED** [review]

**Review by graddaniel — CHANGES_REQUESTED** [review]
Some hints for now, however let's hold off the `name` changes until I talk to management about that.

Also I'll review `app.js` once the library is there.

File: payload-functions.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+function prepareCreateNatGatewayPayload(params) {
+  return {
+    ...(_.pick(params, ["SubnetId", "AllocationId", "DryRun"])),
+    TagSpecifications: helpers.buildTagSpecification("natgateway", params.Tags),
+  };
+}
+
+function validateAssociateRouteTableParams(params) {
+  if (!params.SubnetId && !params.GatewayId) {
+    throw new Error("You need to provide a Subnet ID or a Gateway ID!");
```
Comment (graddaniel) [inline payload-functions.js:69]: `config.json`?

File: payload-functions.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+  }
+}
+
+function prepareAssociateRouteTableToSubnetPayload(params) {
+  validateAssociateRouteTableParams(params);
+  if (!params.SubnetId) {
```
Comment (graddaniel) [inline payload-functions.js:75]: `config.json`?

File: payload-functions.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    throw new Error("Subnet ID is missing!");
+  }
+  return _.omit(params, "GatewayId");
+}
+
+function prepareAssociateRouteTableToGatewayPayload(params) {
+  validateAssociateRouteTableParams(params);
+  if (!params.GatewayId) {
```
Comment (graddaniel) [inline payload-functions.js:83]: `config.json`?

**Review by graddaniel — CHANGES_REQUESTED** [review]
That and the things I've mentioned in our direct conversation:
-Putting `call...Method` call directly where it's used, because we barely reuse it.
-Separation of simple and aggregate AWS functions.

File: config.json [hunk 5]
```diff
@@ -293,18 +298,141 @@
...
       ]
     },
     {
-      "name": "describeKeyPairs",
-      "viewName": "Describe Key Pairs",
+      "name": "describeInstances",
+      "viewName": "Describe Instances",
+      "params": [
+        {
+          "name": "AWS_ACCESS_KEY_ID",
```
[the remark this answers is not in the document]
Comment (graddaniel) [reply config.json:325]: I take that back. We should fix the names some day, but currently that would make the users lose their configs.

File: payload-functions.js (hunk 2 above)
Comment (kdebowski-kaholo) [reply payload-functions.js:69]: Cannot be done in config.json -- either Subnet ID or Gateway ID need to be provided, none of those are strictly required, just one of them is.

File: payload-functions.js (hunk 3 above)
Comment (kdebowski-kaholo) [reply payload-functions.js:75]: same case

File: payload-functions.js (hunk 4 above)
Comment (kdebowski-kaholo) [reply payload-functions.js:83]: same case

**Review by graddaniel — APPROVED** [review]

**Review by graddaniel — DISMISSED** [review]

**Pull request merged**